torrent.go

```diff
@@ -1,13 +1,146 @@
 package tpb
 
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"text/template"
+	"time"
+)
+
 // Torrent represents a Torrent
 type Torrent struct {
-	Name        string
-	Leechers    int
-	Seeders     int
-	User        string
-	Magnet      string
-	Size        uint64
-	Category    TorrentCategory
-	SubCategory TorrentCategory
+	ID          int             `json:"id"`
+	Category    TorrentCategory `json:"category"`
+	Status      UserStatus      `json:"status"`
+	Name        string          `json:"name"`
+	NumFiles    int             `json:"num_files"`
+	Size        uint64          `json:"size"`
+	Seeders     int             `json:"seeders"`
+	Leechers    int             `json:"leechers"`
+	User        string          `json:"username"`
+	Added       time.Time       `json:"added"`
+	Description string          `json:"descripiton"`
+	InfoHash    string          `json:"info_hash"`
+	ImdbID      string          `json:"imdb"`
+	// Magnet      string          `json:"magnet"`
```

Comment:
commented

Reply:
Yep, because I'm still unsure about what to do with the Magnet, should it be inside the Torrent struct, or in a method on the Torrent object as it requests some processing?